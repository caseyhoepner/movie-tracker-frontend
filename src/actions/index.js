export const addMovies = (movies) => ({
  type: 'ADD_MOVIES',
  movies
})

export const toggleFavorite = (movieId) => ({
  type: 'TOGGLE_FAVORITE',
  movieId: movieId
})

export const setUser = (userId) => ({
  type: 'SET_USER',
  userId: userId
})

export const removeUser = () => ({
  type: 'REMOVE_USER'
})

export const filterMovies = (movies) => ({
  type: 'FILTER_MOVIES',
  movies
})