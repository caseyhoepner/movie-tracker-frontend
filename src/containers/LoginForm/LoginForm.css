form {
  height: 300px;
  width: 80%;
  display: flex;
  flex-flow: wrap column;
  justify-content: space-around;
  align-items: center;
  margin: 40px auto;
  letter-spacing: 0.5px;
}

h2 {
  font-size: 2rem;
}

h3 {
  font-size: 1.25rem;
}

.input-container {
  height: 75px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  padding-top: 15px;
}

input,
button {
  font-family: Futura, sans-serif;
  font-size: 1rem;
  height: 1.5rem;
  width: 200px;
}

input {
  padding-left: 5px;
}

button {
  border-width: 2px;
  height: 1.75rem;
  width: 208px;
  color: #D4AA32;
  letter-spacing: 1px;
}

button:hover {
  border: 2px #48788C;
  background-color: #D4AA32;
  color: #48788C;
}

button:active {
  border: 2px #C8901D;
  background-color: #C8901D;
}

button:focus {
  outline: none;
}

.submit-login {
  border-color: #48788C;
  background-color: #48788C;
}

.sign-up-container {
  height: 75px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-around;
  color: #ffffff;
}

.sign-up-btn {
  color: #ffffff;
  border-color: #ffffff;
  background-color: transparent;
}