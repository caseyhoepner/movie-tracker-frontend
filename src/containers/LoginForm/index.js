import React, { Component } from 'react'

export class LoginForm extends Component {
  constructor() {
    super()
    this.state = {

    }
  }

  render() {
    return(
      <form>
        <input />
      </form>
    )
  }
}

export default LoginForm;