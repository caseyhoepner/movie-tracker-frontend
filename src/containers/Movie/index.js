import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { NavLink, Link } from 'react-router-dom';

import * as API from '../../utils'
import { toggleFavorite } from '../../actions';
import './Movie.css';

// make functional
export class Movie extends Component {
  handleClick = async () => {
    const { movie_id, overview, poster_path, release_date, title, vote_average, isLoggedIn, toggleFavorite } = this.props;
    const movie = {
      user_id: isLoggedIn,
      movie_id: movie_id,
      overview,
      poster_path,
      release_date,
      title,
      vote_average,
    }

    await API.addFavorite(movie)
    toggleFavorite(movie_id)
  }

  render() {
    let favoriteBtn;
    const { poster_path, movie_id, isLoggedIn, isFavorite } = this.props;

    if (isLoggedIn) {
      favoriteBtn =
        <button className='m-fav-container'
          onClick={this.handleClick}
        >
          <p className='m-fav-text'>FAVORITE</p>
          <div
            className={`m-fav-btn ${isFavorite ? 'red' : ''}`}>
          </div>
        </button>
    } else {
      favoriteBtn =
        <NavLink className='m-fav-container'
          to='/login'>
          <p className='m-fav-text'>FAVORITE</p>
          <div className='m-fav-btn'>
          </div>
        </NavLink>
    }

    return (
      <div className='m-movie'>
        <Link to={`/movie/${movie_id}`}>
          <div className='m-movie-poster'>
            <img
              className='m-poster'
              src={`http://image.tmdb.org/t/p/w342//${poster_path}`}
              alt='Click here for movie details.'/>
          </div>
        </Link>
        { favoriteBtn }
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  isLoggedIn: state.user.id,
})

export const mapDispatchToProps = (dispatch) => ({
  toggleFavorite: (movieId) => dispatch(toggleFavorite(movieId)),
})

Movie.propTypes = {
  movie_id: PropTypes.number.isRequired,
  overview: PropTypes.string.isRequired,
  poster_path: PropTypes.string.isRequired,
  release_date: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  vote_average: PropTypes.number,
  isFavorite: PropTypes.bool.isRequired,
}

export default connect(mapStateToProps, mapDispatchToProps)(Movie);