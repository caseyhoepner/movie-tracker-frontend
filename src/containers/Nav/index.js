import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { removeUser } from '../../actions';
import userIcon from '../../assets/images/margot.png';
import './Nav.css';

export const Nav = (props) => {
  const { userLoggedIn, removeUser } = props

  const handleHover = (event) => {
    if(!event.target.classList.contains('active')) {
      console.log(event.target.classList)
      event.target.classList.add('active')
    } else {
      console.log(event.target.classList)
      event.target.classList.remove('active')
    }
  }

  return (
    <nav className="nav-container">
      <NavLink
        to="/favorites"
        className="fav-btn-group"
      >
        <div
          className="fav-btn-nav"
          onMouseOver={handleHover}
          onMouseOut={handleHover}
        >
          <span className="num-favs">0</span>
          <p
            className={`favorites-text`}
          >
            FAVORITES
          </p>
        </div>
      </NavLink>
      <NavLink
        to="/login"
        className={userLoggedIn ? "hidden" : "show"}
      >
        <div
          className="login-btn"
          onMouseOver={handleHover}
          onMouseOut={handleHover}
        >
          <img
            className="login-icon"
            src={ userIcon }
            alt='Click here to login'/>
          <p className="login-text">
            LOGIN
          </p>
        </div>
      </NavLink>
      <button
        className={userLoggedIn ? "logout-btn" : "hidden"}
        onClick={removeUser}
      >
        <img
          className="login-icon"
          src={ userIcon }
          alt='Click here to login'/>
        <p className="login-text">
          LOGOUT
        </p>
      </button>
    </nav>
  )
}

export const mapStateToProps = (state) => ({
  userLoggedIn: state.user.id
})

export const mapDispatchToProps = (dispatch) => ({
  removeUser: () => dispatch(removeUser())
})

Nav.propTypes = {
  removeUser: PropTypes.func.isRequired,
  userLoggedIn: PropTypes.number
}

export default connect(mapStateToProps, mapDispatchToProps)(Nav);