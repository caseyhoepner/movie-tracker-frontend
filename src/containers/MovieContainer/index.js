import React from 'react';
import PropTypes from 'prop-types';
import './MovieContainer.css';
import Movie from '../../components/Movie';
import { connect } from 'react-redux';


export const MovieContainer = (props) => {
  const movies = props.movies.map(movie => {
    return <Movie { ...movie } key={movie.id} />
  })

  return (
    <div className='movie-container'>
      { movies }
    </div>
  )
}

const mapStateToProps = (state) => ({
  movies: state.movies
})

MovieContainer.propTypes = {
  movies: PropTypes.array.isRequired
}

export default connect(mapStateToProps)(MovieContainer);