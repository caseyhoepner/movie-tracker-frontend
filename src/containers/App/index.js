import React, { Component } from 'react';
import { Link, Route, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import * as API from '../../utils';
import apiKey from '../../API-key.js';
import { addMovies, toggleFavorite } from '../../actions';
import MovieContainer from '../MovieContainer';
import Nav from '../Nav';
import LoginForm from '../LoginForm'
import MovieDetails from '../../components/MovieDetails';
import wesIcon from '../../assets/images/wes.png';
import './App.css';

export class App extends Component {
  async componentDidMount() {
    const movies = await API.fetchMovies(apiKey)
    await this.props.addMovies(movies)
  }

  render() {
    const { user } = this.props
    return (
      <div className="a-App">
        <header className="a-header-container">
          <Link to='/' className="a-logo">
            <img
              src={ wesIcon }
              className="a-logo-icon"
              alt="An icon of Wes Anderson's face"
            />
            <h1 className='a-header-title'>WesTracker</h1>
          </Link>
          <Nav />
        </header>
        <Route
          exact path='/'
          component={ MovieContainer }
        />
        <Route
          exact path='/login'
          component={ LoginForm }
        />
        <Route path='/movie/:id' render={({match}) => {
          const { id } = match.params;
          const movie = this.props.movies.find(movie => (
            movie.id === parseInt(id, 10)))
          return (
            <MovieDetails {...movie} />
          )}} />
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  user: state.user,
  movies: state.movies
})

export const mapDispatchToProps = (dispatch) => ({
  addMovies: (movies) => dispatch(addMovies(movies)),
  toggleFavorite: (movieId) => dispatch(toggleFavorite(movieId))
})

App.propTypes = {
  user: PropTypes.object.isRequired,
  addMovies: PropTypes.func.isRequired
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(App));