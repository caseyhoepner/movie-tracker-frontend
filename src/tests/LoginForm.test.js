import React from 'react'
import { shallow, mount } from 'enzyme'
import { mockEmail, mockId } from './testMocks';
import { LoginForm, mapDispatchToProps } from '../containers/LoginForm'
import * as Actions from '../actions'

describe('LoginForm', () => {
  describe('LoginFormComponent', () => {
    let wrapper;
    let mockSetUser;

    beforeEach(() => {
      mockSetUser = jest.fn()
      wrapper = shallow(<LoginForm setUser={mockSetUser}/>);
    })

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('state should match the snapshot', () => {
      expect(wrapper.state()).toMatchSnapshot()
    })

    it('should call handleInputChange when email input is changed', () => {
      const spy = spyOn(wrapper.instance(), 'handleInputChange')
      wrapper.instance().forceUpdate()
      const mockEvent = { target: { value: 'something'} }

      wrapper.find('.email-login').simulate('change', mockEvent)

      expect(spy).toHaveBeenCalled()
    })

    it('should call handleInputChange when password input is changed', () => {
      const spy = spyOn(wrapper.instance(), 'handleInputChange')
      wrapper.instance().forceUpdate()
      const mockEvent = { target: { value: 'something'} }

      wrapper.find('.password-login').simulate('change', mockEvent)

      expect(spy).toHaveBeenCalled()
    })

    it('should update state when handleInputChange is called', () => {
      const mockEvent = { target: { name: 'email', value: mockEmail} }

      wrapper.instance().handleInputChange(mockEvent)
      expect(wrapper.state('email')).toBe(mockEmail)
    })

    it('calls handleSubmit onSubmit of the form', () => {
      const spy = spyOn(wrapper.instance(), 'handleSubmit')
      wrapper.instance().forceUpdate()

      wrapper.find('form').simulate('submit')

      expect(spy).toHaveBeenCalled()
    })

    it('should call toggleSigningUp when sign up button is clicked', () => {
      const spy = spyOn(wrapper.instance(), 'toggleSigningUp')
      wrapper.instance().forceUpdate()

      wrapper.find('.lf-sign-up-btn').simulate('click')

      expect(spy).toHaveBeenCalled()
    })

    it('should update state when toggleSigningUp is called', () => {
      expect(wrapper.state('isSigningUp')).toBeFalsy()
      const mockEvent = { preventDefault: jest.fn() }

      wrapper.instance().toggleSigningUp(mockEvent)

      expect(wrapper.state('isSigningUp')).toBeTruthy()

      wrapper.instance().toggleSigningUp(mockEvent)

      expect(wrapper.state('isSigningUp')).toBeFalsy()
    })

    it('should toggle login view when sign up button is clicked', () => {
      const mockEvent = { preventDefault: jest.fn() }

      wrapper.instance().toggleSigningUp(mockEvent)

      expect(wrapper).toMatchSnapshot()
    })

    it('should call addUser if isSigningUp', async () => {
      const spy = spyOn(wrapper.instance(), "createNewUser")
      const mockEvent = { preventDefault: jest.fn() }

      wrapper.instance().setState({ isSigningUp: true })
      await wrapper.instance().handleSubmit(mockEvent);

      expect(spy).toHaveBeenCalled();
    })

    it('should call loginUser if not isSigningUp', async () => {
      const spy = spyOn(wrapper.instance(), "loginUser")
      const mockEvent = { preventDefault: jest.fn() }

      await wrapper.instance().handleSubmit(mockEvent);

      expect(spy).toHaveBeenCalled();
    })
  })

  describe('mapDispatchToProps', () => {
    it('should call dispatch with the addUser action', () => {
      const mockDispatch = jest.fn()
      const expected = Actions.setUser(mockId)

      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.setUser(mockId)

      expect(mockDispatch).toHaveBeenCalledWith(expected)
    })
  })
})