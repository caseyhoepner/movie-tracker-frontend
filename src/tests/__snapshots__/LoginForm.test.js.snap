// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm LoginFormComponent should match the snapshot 1`] = `
<div>
  <div
    className="hidden"
  />
  <form
    className="lf-form"
    onSubmit={[Function]}
  >
    <h2
      className="banner-text"
    >
      HAVE AN ACCOUNT?
    </h2>
    <div
      className="input-container"
    >
      <input
        className="hidden"
        name="name"
        onChange={[Function]}
        placeholder="Klaus"
        value=""
      />
      <input
        className="email-login"
        name="email"
        onChange={[Function]}
        placeholder="wes@anderson.com"
        type="email"
        value=""
      />
      <input
        className="password-login"
        name="password"
        onChange={[Function]}
        placeholder="Type password here"
        type="password"
        value=""
      />
    </div>
    <button
      className="submit-login"
      disabled={true}
      onClick={[Function]}
    >
      LOGIN
    </button>
    <div
      className="sign-up-container"
    >
      <h3
        className="lower-banner-text"
      >
        NEED AN ACCOUNT?
      </h3>
      <button
        className="sign-up-btn"
        onClick={[Function]}
      >
        SIGN UP
      </button>
    </div>
  </form>
</div>
`;

exports[`LoginForm LoginFormComponent should toggle login view when sign up button is clicked 1`] = `
<div>
  <div
    className="hidden"
  />
  <form
    className="lf-form"
    onSubmit={[Function]}
  >
    <h2
      className="banner-text"
    >
      ENLIST IN TEAM ZISSOU
    </h2>
    <div
      className="input-container"
    >
      <input
        className="name-login"
        name="name"
        onChange={[Function]}
        placeholder="Klaus"
        value=""
      />
      <input
        className="email-login"
        name="email"
        onChange={[Function]}
        placeholder="klaus@daimler.net"
        type="email"
        value=""
      />
      <input
        className="password-login"
        name="password"
        onChange={[Function]}
        placeholder="Type password here"
        type="password"
        value=""
      />
    </div>
    <button
      className="submit-login"
      disabled={true}
      onClick={[Function]}
    >
      CREATE ACCOUNT
    </button>
    <div
      className="sign-up-container"
    >
      <h3
        className="lower-banner-text"
      >
        ERM, NVM...
      </h3>
      <button
        className="sign-up-btn"
        onClick={[Function]}
      >
        CANCEL SIGN UP
      </button>
    </div>
  </form>
</div>
`;

exports[`LoginForm LoginFormComponent state should match the snapshot 1`] = `
Object {
  "email": "",
  "errorMessage": "",
  "isDisabled": true,
  "isSigningUp": false,
  "name": "",
  "password": "",
}
`;
